client/components/progress-bar/index.jsx

```diff
@@ -28,14 +29,25 @@ export default class ProgressBar extends PureComponent {
 		compact: PropTypes.bool,
 		className: PropTypes.string,
 		isPulsing: PropTypes.bool,
+		canGoBackwards: PropTypes.bool,
 	};
 
+	static getDerivedStateFromProps( props, state ) {
+		return {
+			allTimeMax: Math.max( state.allTimeMax, props.value ),
+		};
+	}
+
 	state = {
 		allTimeMax: this.props.value,
 	};
 
 	getCompletionPercentage() {
-		const percentage = Math.ceil( this.state.allTimeMax / this.props.total * 100 );
+		const percentage = Math.ceil(
+			( this.props.canGoBackwards ? this.props.value : this.state.allTimeMax ) /
+				this.props.total *
```

Comment:
this ought to have `( this.props.total + Number.EPSILON )` to prevent division-by-zero errors :smile: